### DISCUSSION TITLE
is there any way to place node based on center of node's position?

### DISCUSSION THREAD

**Discussion by jaeeewon** [xyflow/xyflow#2827] [Q&A] [answered]
well.. i usually place node as spherical form.
so.. i got coordinates by using sin & cos.
but.. node didn't place as i have expected.
maybe node is placed based on (0, 0) of node.

![image](https://user-images.githubusercontent.com/87597383/217930400-37c5947c-1e7d-42cc-bb2a-58cb1a47b49d.png)

this won't be problem if rectangle is square.
but.. my node isn't..

under is the way i wanna place..
thank you ^--^

![image](https://user-images.githubusercontent.com/87597383/217929714-42ffdc7b-ae9c-4f36-a43d-fa28d0cd5561.png)

**Comment by moklick** [comment, marked as answer]
You can use the `nodeOrigin` prop. By default it's `[0, 0]`. You want to set it to `[0.5, 0.5]`. https://reactflow.dev/docs/api/react-flow-props/#nodeorigin

**Reply by jaeeewon** [reply]
Appreciate a lot :)

**Reply by rickyzhangca** [reply]
is it possible to define `nodeOrigin` per node...? thanks
